ktor-shared/ktor-serialization/common/src/ContentConverter.kt

```diff
@@ -119,8 +119,7 @@ public suspend fun List<ContentConverter>.deserialize(
     // 1. there is no suitable converter
     // 2. result of deserialization is null
     // We can differentiate these cases by checking if body was consumed or not
-    val result = this.asFlow()
-        .filter { !body.isClosedForRead }
```

Comment:
This check is needed when some converter has consumed body and returned `null`. The next converter will crash with EOF if it tries to parse.

Reply:
you can move it to `firstOrNull { it != null || body.isClosedForRead }`. Now it's confusing.

Reply:
sure, let me check